Sources/Contour/Transport/RabbitMQ/BusConsumerConfigurationEx.cs

```diff
@@ -38,6 +40,17 @@ public static IReceiverConfigurator WithQoS(this IReceiverConfigurator builder,
         }
 
         /// <summary>
+        /// Returns current QoS settings
+        /// </summary>
+        /// <param name="builder"></param>
+        /// <returns></returns>
+        public static Maybe<QoSParams> GetQoS(this IReceiverConfigurator builder)
+        {
+            var qos = ((RabbitReceiverOptions)((ReceiverConfiguration)builder).Options).GetQoS();
```

Comment:
As discussed need to cast safely and check for null. Further refactoring needed in BusConsumerConfigurationEx.